Source/WebCore/html/canvas/CanvasRenderingContext2DBase.cpp

```diff
@@ -2184,9 +2184,26 @@ ExceptionOr<RefPtr<CanvasPattern>> CanvasRenderingContext2DBase::createPattern(C
     return Exception { TypeError };
 }
 
-void CanvasRenderingContext2DBase::didDrawEntireCanvas()
+OptionSet<CanvasRenderingContext2DBase::DidDrawOption> CanvasRenderingContext2DBase::defaultDidDrawOptions()
```

Comment:
I think you should be able to put this function at the top of the file as a free function:
```
static OptionSet<DidDrawOption> defaultDidDrawOptions() 
{
   return  { DidDrawOption::ApplyTransform, DidDrawOption::ApplyShadow, DidDrawOption::ApplyClip, DidDrawOption::ApplyPostProcessing };
}
```

No static locals, OptionSet reduces to one constexpr number.

Reply:
Keeping them as static members allows us to set default method parameters values in the header, otherwise I would follow your suggestion.